src/components/Signup/Signup.jsx

```diff
@@ -0,0 +1,255 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { Link } from 'react-router-dom';
+import Joi from 'joi-browser';
+import PropTypes from 'prop-types';
+import signupInputSchema from '../../schemas/signup';
+import signupAction from '../../actions/user/signupAction';
+import Form from '../Form/Form';
+import Input from '../Input/Input';
+import Button from '../Button/Button';
+import GoogleIcon from '../../../assets/icon/google.png';
+import TwitterIcon from '../../../assets/icon/twitter.png';
+import FacebookIcon from '../../../assets/icon/facebook.png';
+import './Signup.scss';
+
+/**
+ *
+ *
+ * @export
+ * @class Signup
+ * @extends {Component}
+ */
+export class Signup extends Component {
+  state = {
+    data: {
+      firstName: '',
+      lastName: '',
+      userName: '',
+      email: '',
+      password: '',
+      confirmPassword: '',
+    },
+    errors: {},
+  };
+
+  getValidationErrorMessages = (error) => {
+    const validationErrors = {};
+    const errorDetails = error.details;
+
+    for (let i = errorDetails.length - 1; i >= 0; i -= 1) {
```

Comment:
each time the loop runs `errorDetails.length` will be evaluated, why not get the size first before the loop starts.

Reply:
I have implemented this feedback too, @dinobi